### ISSUE TITLE
How to contribute

### ISSUE THREAD

**Issue by waldyrious** [neutralitywtf/ConceptReplacer#13]
The first example featured in the home page is currently [Ada Lovelace](http://www.neutrality.wtf/?url=https%3A%2F%2Fen.wikipedia.org%2Fwiki%2FAda_Lovelace), and right in the first sentence there's "Countess of Lovelace", which should have been switched to "Count of Lovelace".

First of all, it wasn't immediately clear where the code repository for this experiment was located, and second, after I came here I wasn't sure how to propose the change.

Should I edit `men.json` to change `"singular": [ "king" ],` into `"singular": [ "king" , "count" ],` then edit `women.json` to change `"singular": [ "queen" ],` into `"singular": [ "queen" , "countess" ],`? Or should I edit the `royalty` key instead? Should I create a new key, otherwise? And if so, what would be the appropriate nomenclature?

These questions make it hard for interested people to contribute and make the experiment less useful as a consequence. IMO some instructions in the README, or a CONTRIBUTING.md file (and a more prominent link in the website) would help.

/cc @slashme

**Comment by mooeypoo** [comment]
There's a problem adding "count" to the dictionary -- the word is also a verb (I count numbers, point count, etc) and since the replacer is "dumb" (just plain replacements) it will not recognize the difference between the count to replace and the verb.

While the replacement is plain/dumb and direct, there will be some words that we won't be able to include because the replacement will do more "harm" than good (replacing "count" as in "counting" with "Countess" will be more disruptive than leaving it alone.)

The words are in the dictionaries and the dictionaries must have the same categories -- but you have a point about the explanation. I'll add a full description + instructions on how to suggest more words to the README.

One of the reasons this was delayed a bit was because I wanted to create a quick script to actually print the words + categories (from both dictionaries) in a table, so it's easier to compare and see. But the explanation should be there regardless. Thanks for pointing that out!

**Comment by waldyrious** [comment]
Understood. By the way, Wikidata Game has an [interesting approach](https://bitbucket.org/magnusmanske/wikidata-game/src/ce73dc35f07d5d6d56b5823c0bc0081b7dbd8bd3/public_html/main.js?at=master#main.js-932) to this, using regular expressions to determine which words carry gender information, depending on the surrounding context, capitalization, punctuation, etc.

Is that something that this project could consider down the road?

**Comment by mooeypoo** [comment]
Absolutely! I am also looking into grammar libraries that analyze words by their grammatical type (noun vs verb vs adjective, etc) that can open the door for even better replacements. This will be in the future, though...

I'll take a look at how wikidata does it, but I think it might be more of a challenge to try to "seemlessly replace" words (leaving the language intact and making sense) than analyzing whether a text has female/male terminology.

But yes, I'd love to look into this for a future version, for sure!

**Comment by mooeypoo** [comment]
By the way, a previous version of the tool did use regular expressions, but that still didn't quite solve the issue of recognizing words like count (counting something) vs count (the title). Also, it was a lot harder making the dictionary bidirectional, where you can use either of the dictionaries to either search *or* replace.

Maybe this should be done differently in a future version, and the dictionaries should be split to search terms and replaced terms, but the idea here was to try and work with dictionaries that can just be combined or replaced interchangeably. That way, you can also add other modules and switch it up, like adding a dictionary for neutral gender (they / theirs) and replacing both men and women terms with it, etc.

In any case, this is definitely an interesting problem to look at. Thanks for raising it :)

**Comment by waldyrious** [comment]
Great points. I'm looking forward to see how this project evolves, and hopefully I'll be able to contribute :)
